### PR TITLE
Introduce new domain objects to match the updated schema. Update the UI.

### PR DESCRIPTION
**Pull request by seanlip**
PTAL; further changes are somewhat blocked on this since this sets up the basic data objects in the frontend.

Planned next steps:
- Support multiple prompts in the UI.
- Get the overall feedback flow working.

### CODE REVIEW

File: client/domain/PerformanceTestObjectFactory.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    var PerformanceTest = function(performanceTestDict) {
+      this._inputDataAtom = performanceTestDict.input_data_atom;
+      this._transformationFunction =
+        performanceTestDict.transformation_function;
+      this._expectedPerformance = performanceTestDict.expected_performance;
+      this._evaluationFunction = performanceTestDict.evaluation_function;
+    };
+
+    // Instance methods.
+    PerformanceTest.prototype.getTransformationFunction = function() {
```
Comment (eyurko) [inline client/domain/PerformanceTestObjectFactory.js]: We'll also need getEvaluationFunction, right?

File: client/services/QuestionDataService.js [hunk 2]
```diff
@@ -21,10 +21,19 @@ tie.factory('QuestionDataService', [
...
     var _DATA_DICT = {
-      "language": "python",
-      "stages": [{
-        "instructions": "Welcome to this programming exercise. Implement the abbreviate function.",
-        "data": [{
+      "title": "Internationalization (i18n)",
+      "prompts": [{
+        "instructions": [
+          "Welcome to this programming exercise.",
+          "You will complete this exercise by editing the code in the sample.py file using your favorite editor. Every time you save the file, I will evaluate your code and let you know about next steps.",
```
Comment (eyurko) [inline client/services/QuestionDataService.js]: Drop this sentence, then, given the suggested changes to rle.json

Comment (seanlip) [inline client/services/QuestionDataService.js]: Done.

File: client/domain/PerformanceTestObjectFactory.js (hunk 1 above)
Comment (seanlip) [inline client/domain/PerformanceTestObjectFactory.js]: Discussed with @eyurko; this is a no-op. We plan to add other functions lazily to the domain objects as we need them, in order to minimize the amount of boilerplate and to make it less work to handle a schema change while we're still in the process of fleshing things out.

**Comment by seanlip** [comment]
Thanks, @eyurko! Comments addressed; PTAL.

**Pull request merged**